pkg/drivers/powerflex.go

```diff
@@ -371,28 +377,41 @@ func ExtractZonesFromSecret(ctx context.Context, kube client.Client, namespace s
 		yamlConfig := make([]StorageArrayConfig, 0)
 		configs, err := yaml.JSONToYAML(configBytes)
 		if err != nil {
-			return nil, fmt.Errorf("unable to parse multi-array configuration[%v]", err)
+			return fmt.Errorf("unable to parse multi-array configuration %v", err)
 		}
 		err = yaml.Unmarshal(configs, &yamlConfig)
 		if err != nil {
-			return nil, fmt.Errorf("unable to unmarshal multi-array configuration[%v]", err)
+			return fmt.Errorf("unable to unmarshal multi-array configuration %v", err)
 		}
 
+		var labelKey string
+		var numArrays, numArraysWithZone int
+		numArrays = len(yamlConfig)
 		for _, configParam := range yamlConfig {
 			if configParam.SystemID == "" {
-				return nil, fmt.Errorf("invalid value for SystemID")
+				return fmt.Errorf("invalid value for SystemID")
+			}
+			if labelKey == "" {
+				labelKey = configParam.Zone.LabelKey
```

Comment:
should we check if configParam.Zone exists first?

Reply:
@JacobGros In this case I believe Zone itself will be empty if it doesn't exist, since it is defined as value above and not a reference.

Reply:
@JacobGros we also have a unit-test for this case that covers the scenario where zone does not exist.